src/main/java/bio/terra/service/snapshot/flight/delete/SnapshotDeleteFlight.java

```diff
@@ -137,6 +137,12 @@ public SnapshotDeleteFlight(FlightMap inputParameters, Object applicationContext
     addStep(new PerformAzureStep(new DeleteSnapshotMetadataAzureStep(azureStorageAccountService)));
     addStep(new PerformSnapshotStep(new UnlockSnapshotStep(snapshotDao, snapshotId)));
 
+    // delete snapshot container
+    addStep(
+        new PerformAzureStep(
+            new DeleteSnapshotDeleteStorageAccountStep(
+                snapshotId, resourceService, tableDao, profileService)));
+
```

Comment:
Ah, that's because it needs to run _after_ the snapshot metadata is removed from postgres for that the FK isn't violated on delete (since the snapshot table references the storage account table)